amptorch/dataset_lmdb.py

```diff
@@ -0,0 +1,365 @@
+                target_scaler_list.append(temp_target_scaler)
+                descriptor_setup_list.append(temp_descriptor_setup)
+                descriptor_list.append(temp_descriptor)
+                elements_list.append(temp_elements)
+
+        self._keylen_cumulative = np.cumsum(self.length_list).tolist()
+        self.total_length = np.sum(self.length_list)
+        self.feature_scaler = feature_scaler_list[0]
+        self.target_scaler = target_scaler_list[0]
+        self.descriptor_setup = descriptor_setup_list[0]
+        self.descriptor = descriptor_list[0]
+        self.elements = elements_list[0]
+        self.loaded_db_idx = -1
+
+        if len(self.db_paths) > 1:
+            if any(
+                feature_scaler != self.feature_scaler
+                for feature_scaler in feature_scaler_list
+            ):
+                raise ValueError(
+                    "Please make sure all lmdb used the same feature scaler"
+                )
+            if any(
+                target_scaler != self.target_scaler
+                for target_scaler in target_scaler_list
+            ):
+                raise ValueError(
+                    "Please make sure all lmdb used the same target scaler"
+                )
+            if any(descriptor != self.descriptor for descriptor in descriptor_list):
+                raise ValueError("Please make sure all lmdb used the same descriptor")
+            if any(set(elements) != set(self.elements) for elements in elements_list):
+                raise ValueError("Please make sure all lmdb used the same elements")
+
+    def __len__(self):
+        return self.total_length
+
+    def __getitem__(self, idx):
+        db_idx = bisect.bisect(self._keylen_cumulative, idx)
+        if db_idx != 0:
+            el_idx = idx - self._keylen_cumulative[db_idx - 1]
+        else:
+            el_idx = idx
+
+        with self.envs[db_idx].begin(write=False) as txn:
+            data = txn.get(self.keys_list[db_idx][el_idx])
+            data_object = pickle.loads(data)
+
+        return data_object
+
+    def get_descriptor(self, descriptor_setup):
+        fp_scheme, fp_params, cutoff_params, elements = descriptor_setup
+        if fp_scheme == "gaussian":
+            descriptor = Gaussian(Gs=fp_params, elements=elements, **cutoff_params)
+        elif fp_scheme == "gmp":
+            descriptor = GMP(MCSHs=fp_params, elements=elements)
+        else:
+            raise NotImplementedError
+        return descriptor
+
+    @property
+    def input_dim(self):
+        return self[0].fingerprint.shape[1]
+
+    def connect_db(self, lmdb_path):
+        env = lmdb.open(
+            lmdb_path,
+            subdir=False,
+            readonly=True,
+            lock=False,
+            readahead=False,
+            meminit=False,
+            max_readers=1,
+        )
+        return env
+
+
+class AtomsLMDBDatasetPartialCache(Dataset):
```

Comment:
For all the LMDB variants can you add a brief description under each one to highlight their differences please?